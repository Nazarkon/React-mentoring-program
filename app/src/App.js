import React, { lazy, Suspense } from 'react';
import './App.scss';
import MovieListPage from './components/MovieListPage/MovieListPage';
import { BrowserRouter, Routes, Route } from 'react-router-dom';

const MovieDetails = lazy(() => import('./components/MovieDetails/MovieDetails'));

function App() {
  return (
    <BrowserRouter>
      <Routes>
        <Route path="/" element={<MovieListPage />}>
          <Route
            path="/movie-details/:id"
            element={
              <Suspense fallback={<div>Loading...</div>}>
                <MovieDetails />
              </Suspense>
            }
          />
        </Route>
      </Routes>
    </BrowserRouter>
  );
}

export default App;
