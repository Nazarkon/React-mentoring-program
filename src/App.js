import Counter from "./components/Counter/Counter"
import SearchInput from './components/SearchInput/SearchInput';
import GenreList from './components/GenreList/GenreList';
import genreList from "./mock/genreData.json"
import { useState } from 'react';

function App() {
  const [genre, setSelectedGenre] = useState(genreList[0])

  const searchValue = (value) => {
    console.log(`Search value: ${value}`)
  }

  return (
    <div className="App">
     <Counter
     defaultValue={2}
     />
     <SearchInput
     defaultValue="Search something"
     onSearch={searchValue}
     />
     <GenreList 
        genreList={genreList}
        currentItem={genre}
        updateList={setSelectedGenre}
      />
    </div>
  );
}

export default App;
