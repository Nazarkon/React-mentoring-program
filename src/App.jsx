/* eslint-disable no-unused-vars */
import React, { lazy, Suspense, useState } from 'react';
import axios from 'axios';
import './App.scss';
import { BrowserRouter, Routes, Route } from 'react-router-dom';
import MovieListPage from './components/MovieListPage/MovieListPage';

const MovieDetails = lazy(() => import('./components/MovieDetails/MovieDetails'));
const AddMovieForm = lazy(() => import('./components/AddMovieForm/AddMovieForm'));
const SearchInput = lazy(() => import('./components/SearchInput/SearchInput'));

function App() {
  const [searchParams, setSearchParams] = useState();

  const handleSearch = (searchParamsFromUrl) => {
    setSearchParams(searchParamsFromUrl);
  };
  return (
    <BrowserRouter>
      <Routes>
        <Route
          path="/"
          element={<MovieListPage searchParamsURL={searchParams} />}
        >
          <Route
            path="/"
            name="searchInput"
            element={(
              <Suspense fallback={<div>Loading...</div>}>
                <SearchInput onSearch={handleSearch} />
              </Suspense>
            )}
          >
            <Route path="/new" name="addFilm" element={<AddMovieForm />} />
            <Route
              path="/delete/:id"
              name="deleteFilm"
              element={<AddMovieForm />}
            />
            <Route
              path="/edit/:id"
              name="editFilm"
              element={<AddMovieForm />}
            />
          </Route>
          <Route
            path="/movie-details/:id"
            name="movieDetails"
            element={(
              <Suspense fallback={<div>Loading...</div>}>
                <MovieDetails />
              </Suspense>
            )}
          />
          <Route
            path="/new"
            name="addMovieForm"
            element={(
              <Suspense fallback={<div>Loading...</div>}>
                <AddMovieForm />
              </Suspense>
            )}
          />
        </Route>
      </Routes>
    </BrowserRouter>
  );
}

export default App;
