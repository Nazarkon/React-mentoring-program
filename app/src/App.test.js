import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';

import App from './App';

describe('Check that all components presented on a page', () => {
  test('List of items exist', () => {
    render(<App />);
    const listElement = screen.getByRole('list');
    expect(listElement).toBeInTheDocument();
  });
});
