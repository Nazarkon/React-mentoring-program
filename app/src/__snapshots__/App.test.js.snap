// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component matches snapshot 1`] = `
<div>
  <header
    class="header-container"
  >
    <p
      class="header-title"
    >
      <strong
        class="header-title-first"
      >
        netflix
      </strong>
      roulette
    </p>
    <button
      class="header-button"
      type="button"
    >
      + Add Movie
    </button>
  </header>
  <div
    class="movie-page-container"
  >
    <div
      class="movie-page-search-image-background"
    >
      <div>
        Loading...
      </div>
    </div>
    <div>
      <div
        class="movie-page-film-filter"
      >
        <ul
          class="list-container"
        >
          <li
            class="item-light"
          >
            All
          </li>
          <li
            class=""
          >
            Fantasy
          </li>
          <li
            class=""
          >
            Adventure
          </li>
          <li
            class=""
          >
            Romance
          </li>
          <li
            class=""
          >
            Science Fiction
          </li>
          <li
            class=""
          >
            Documentary
          </li>
          <li
            class=""
          >
            Comedy
          </li>
          <li
            class=""
          >
            Action
          </li>
        </ul>
        <label
          class="select-label"
        >
          <span
            class="select-label-text"
          >
            SORT BY:
          </span>
          <select
            class="select-component"
          >
            <option
              class="select-option"
              value="release_date"
            >
              RELEASE DATE
            </option>
            <option
              class="select-option"
              value="title"
            >
              TITLE
            </option>
          </select>
        </label>
      </div>
      <div
        class="movie-page-film-counter"
      >
        <strong>
          0
        </strong>
         
        movies found
      </div>
      <div
        class="movie-page-film-list"
      />
    </div>
  </div>
  <footer
    class="footer-container"
  >
    <p
      class="footer-title"
    >
      <strong
        class="footer-title-first"
      >
        netflix
      </strong>
      roulette
    </p>
  </footer>
</div>
`;
